app/src/main/java/fr/free/nrw/commons/contributions/SetWallpaperWorker.java

```diff
@@ -0,0 +1,130 @@
+import com.facebook.drawee.backends.pipeline.Fresco;
+import com.facebook.imagepipeline.core.ImagePipeline;
+import com.facebook.imagepipeline.datasource.BaseBitmapDataSubscriber;
+import com.facebook.imagepipeline.image.CloseableImage;
+import com.facebook.imagepipeline.request.ImageRequest;
+import com.facebook.imagepipeline.request.ImageRequestBuilder;
+import fr.free.nrw.commons.R;
+import java.io.IOException;
+import timber.log.Timber;
+
+public class SetWallpaperWorker extends Worker {
+
+    private static final String NOTIFICATION_CHANNEL_ID = "set_wallpaper_channel";
+    private static final int NOTIFICATION_ID = 1;
+
+    public SetWallpaperWorker(@NonNull Context context, @NonNull WorkerParameters params) {
+        super(context, params);
+    }
+
+    @NonNull
+    @Override
+    public Result doWork() {
+        Context context = getApplicationContext();
+        createNotificationChannel(context);
+        showProgressNotification(context);
+
+        String imageUrl = getInputData().getString("imageUrl");
+        if (imageUrl == null) {
+            return Result.failure();
+        }
+
+        ImageRequest imageRequest = ImageRequestBuilder
+            .newBuilderWithSource(Uri.parse(imageUrl))
+            .build();
+
+        ImagePipeline imagePipeline = Fresco.getImagePipeline();
+        final DataSource<CloseableReference<CloseableImage>>
+            dataSource = imagePipeline.fetchDecodedImage(imageRequest, context);
+
+        dataSource.subscribe(new BaseBitmapDataSubscriber() {
+            @Override
+            public void onNewResultImpl(@Nullable Bitmap bitmap) {
+                if (dataSource.isFinished() && bitmap != null) {
+                    Timber.d("Bitmap loaded from url %s", imageUrl.toString());
+                    setWallpaper(context, Bitmap.createBitmap(bitmap));
+                    dataSource.close();
+                }
+            }
+
+            @Override
+            public void onFailureImpl(DataSource dataSource) {
+                Timber.d("Error getting bitmap from image url %s", imageUrl.toString());
+                showNotification(context, "Setting Wallpaper Failed", "Failed to download image.");
+                if (dataSource != null) {
+                    dataSource.close();
+                }
+            }
+        }, CallerThreadExecutor.getInstance());
+
+        return Result.success();
+    }
+
+    private  void setWallpaper(Context context, Bitmap bitmap) {
+        WallpaperManager wallpaperManager = WallpaperManager.getInstance(context);
+
+        try {
+            wallpaperManager.setBitmap(bitmap);
+            showNotification(context, "Wallpaper Set", "Wallpaper has been updated successfully.");
+
+        } catch (Exception e) {
+            Timber.e(e, "Error setting wallpaper");
+            showNotification(context, "Setting Wallpaper Failed", " "+e.getLocalizedMessage());
+        }
+    }
+
+    private void showProgressNotification(Context context) {
```

Comment:
@nicolas-raoul If we agree on deprecating progress dialog permanently can we create notification methods as an interface method and call it wherever we need. for ex : performing login. 

This will make the code more readable and reusable.

Reply:
That could be discussed Separately I believe, not here

Reply:
Hi Shashank, 

I apologies If I got you wrong. I believe there's value in discussing it over here, especially within the relevant context. My intention was to create an extension utility class to adhere to the DRY (Don't Repeat Yourself) pattern. I included the message to facilitate the creation of an issue (referencing it in a new issue), this comment link would provide additional assistance to contributors in understanding the idea better.